src/Dictionaries/PostgreSQLDictionarySource.cpp

```diff
@@ -68,29 +68,32 @@ PostgreSQLDictionarySource::PostgreSQLDictionarySource(const PostgreSQLDictionar
 BlockInputStreamPtr PostgreSQLDictionarySource::loadAll()
 {
     LOG_TRACE(log, load_all_query);
-    return std::make_shared<PostgreSQLBlockInputStream>(
-            connection->conn(), load_all_query, sample_block, max_block_size);
+    auto tx = std::make_shared<pqxx::read_transaction>(*connection->conn());
+    return std::make_shared<PostgreSQLBlockInputStream<pqxx::read_transaction>>(tx, load_all_query, sample_block, max_block_size);
```

Comment:
> Why do we write something through *InputStream?

It is needed only for initial sync, which is done with specific isolation level and with `set transaction snapshot`, which requires pqxx::work object. So I changed this InputStream class to admit both pqxx::work and pqxx::read_transaction, instead of just a connection object.